### PR TITLE
BUG: fix reindexing with a tz-aware index and method='nearest'

### LINKED ISSUE #26683: Warning while reindexing tz aware index with method='nearest'
Reindexing a tz aware dataframe using method='nearest' raise an internal warning.
```python
from pandas.util.testing import makeTimeDataFrame
df = makeTimeDataFrame(freq='1h')
df = df.tz_localize('UTC')
df.reindex(df.index[1:4], method='nearest')
```
raises a warning:
```
/lib/python3.7/site-packages/pandas/core/indexes/base.py:2820: FutureWarning: Converting timezone-aware DatetimeArray to timezone-naive ndarray with 'datetime64[ns]' dtype. In the future, this will return an ndarray with 'object' dtype where each element is a 'pandas.Timestamp' with the correct 'tz'.
        To accept the future behavior, pass 'dtype=object'.
        To keep the old behavior, pass 'dtype="datetime64[ns]"'.
  target = np.asarray(target)
```
in pandas 0.24.2

### PR DESCRIPTION
**Pull request by kandersolar**
- [x] closes #26683
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

First pandas PR, happy to make changes as needed.  Is a whatsnew entry needed?  And if so, I assume it should go in v1.0.1.rst?

### CODE REVIEW

File: pandas/tests/frame/indexing/test_indexing.py [hunk 1]
```diff
@@ -1599,6 +1599,14 @@ def test_reindex_methods_nearest_special(self):
         actual = df.reindex(target, method="nearest", tolerance=[0.5, 0.01, 0.4, 0.1])
         tm.assert_frame_equal(expected, actual)
 
+    def test_reindex_nearest_tz(self):
```
Comment (MarcoGorelli) [inline pandas/tests/frame/indexing/test_indexing.py]: Thanks!

Can you add the issue number number under the test? So
```
def test_reindex_nearest_tz(self):
    # GH26683
```

(many old tests didn't do this, but it seems to be preferred for new ones)

File: pandas/tests/frame/indexing/test_indexing.py [hunk 2]
```diff
@@ -1599,6 +1599,14 @@ def test_reindex_methods_nearest_special(self):
...
+        idx = pd.date_range("2019-01-01", periods=5, tz="US/Eastern")
```
Comment (MarcoGorelli) [inline pandas/tests/frame/indexing/test_indexing.py]: There are some fixtures defined in `conftest.py`, I think here you might want to add `tz_aware_fixture`:
```
def test_reindex_nearest_tz(self, tz_aware_fixture):
    tz = tz_aware_fixture
    idx = pd.date_range("2019-01-01", periods=5, tz=tz)
```

**Comment by kanderso-nrel** [comment]
Thanks @MarcoGorelli for the guidance!  Is there anything else I can improve?

**Comment by kanderso-nrel** [comment]
Docs build fail seems unrelated, transient network error?  https://github.com/pandas-dev/pandas/pull/31511/checks?check_run_id=419939179#step:6:37

**Comment by charlesdong1991** [comment]
@kanderso-nrel likely a network issue, you could add a whatsnew note in `1.1` and try it again

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -151,7 +151,7 @@ Indexing
 - Bug in :meth:`Series.at` and :meth:`DataFrame.at` not matching ``.loc`` behavior when looking up an integer in a :class:`Float64Index` (:issue:`31329`)
 - Bug in :meth:`PeriodIndex.is_monotonic` incorrectly returning ``True`` when containing leading ``NaT`` entries (:issue:`31437`)
 - Bug in :meth:`DatetimeIndex.get_loc` raising ``KeyError`` with converted-integer key instead of the user-passed key (:issue:`31425`)
--
+- Bug in :meth:`DataFrame.reindex` and :meth:`Series.reindex` when reindexing with a tz-aware index and method='nearest' (:issue:`26683`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst]: can you move to Datetimelike

File: pandas/core/indexes/base.py [hunk 4]
```diff
@@ -3065,9 +3065,8 @@ def _get_nearest_indexer(self, target: "Index", limit, tolerance) -> np.ndarray:
         left_indexer = self.get_indexer(target, "pad", limit=limit)
         right_indexer = self.get_indexer(target, "backfill", limit=limit)
 
-        target = np.asarray(target)
-        left_distances = abs(self.values[left_indexer] - target)
-        right_distances = abs(self.values[right_indexer] - target)
+        left_distances = abs(self[left_indexer] - target)
```
Comment (jreback) [inline pandas/core/indexes/base.py]: can we change to use np.abs

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, some comments, pls merge master. ping on green.

**Comment by kanderso-nrel** [comment]
@jreback all green after #31594

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**